extensions/InputBox/InputBox.hooks.php

```diff
@@ -52,24 +52,38 @@ public static function onMediaWikiPerformAction(
 		$request, 
 		$wiki )
 	{
+		global $wgScript, $wgUser;
+
 		if( $wiki->getAction( $request ) !== 'edit' ){
 			# not our problem
 			return true;
 		}
+
+		$params = $request->getValues();
+
+		if ( $request->getText( 'preload', '' ) !== '' ){
+			# Make sure any call with preload uses VE when it matches the user's prefs
+			if ( EditorPreference::isVisualEditorPrimary() && $wgUser->isLoggedIn() ) {
+				$params['veaction'] = 'edit';
+				unset( $params['action'] );
+
+				$output->redirect( wfAppendQuery( $wgScript, $params ), '301' );
+				return false;
```

Comment:
Yeah, I'd make the early return if neither prefix nor preload are set, i.e.
```php
if ( $request->getText( 'prefix', '' ) === '' || $request->getText( 'preload', '' ) === '' ) {
	// Fine
	return true;
}
```

Then put each of the params changes in if blocks, i.e. something like:
```php
if (
	!empty( $params['preload'] )
	&& EditorPreference::isVisualEditorPrimary()
	&& $user->isLoggedIn()
) {
	$params['veaction'] = 'edit';
	unset( $params['action'] );
}

if ( !empty( $params['prefix'] ) ) {
	$title = $params['prefix'];
	if ( isset( $params['title'] ) ) {
		$title .= $params['title'];
	}
	unset( $params['prefix'] );
	$params['title'] = $title;
}

$output->redirect( wfAppendQuery( $wgScript, $params ), '301' );
return false;
```

What do you think?